searx/plugins/__init__.py

```diff
@@ -77,6 +87,80 @@ def call(self, ordered_plugin_list, plugin_type, request, *args, **kwargs):
         return ret
 
 
+def load_external_plugins(plugin_names):
+    plugins = []
+    for name in plugin_names:
+        logger.debug('loading plugin: {0}'.format(name))
+        try:
+            pkg = import_module(name)
+        except Exception as e:
+            logger.critical('failed to load plugin module {0}: {1}'.format(name, e))
+            exit(3)
+
+        pkg.__base_path = dirname(abspath(pkg.__file__))
+
+        prepare_package_resources(pkg, name)
+
+        plugins.append(pkg)
+        logger.debug('plugin "{0}" loaded'.format(name))
+    return plugins
+
+
+def sync_resource(base_path, resource_path, name, target_dir, plugin_dir):
+    dep_path = join(base_path, resource_path)
+    file_name = basename(dep_path)
+    resource_path = join(target_dir, file_name)
+    if not exists(resource_path) or sha_sum(dep_path) != sha_sum(resource_path):
+        try:
+            copyfile(dep_path, resource_path)
```

Comment:
I would add:
```python
# copy atime_ns and mtime_ns, so the weak ETags (generated by
# the HTTP server) do not change.
dep_stat = os.stat(dep_path)
os.utime(resource_path, ns=(dep_stat.st_atime_ns, dep_stat.st_mtime_ns))
```

Reply:
Great, thanks, mtime modification added.

Reply:
It doesn't work as expected:
* mtime, atime are copied from the site-package directory.
* but setuptools doesn't copy these atributes.

Reply:
pip:
* tar: OK(os.utime): https://github.com/pypa/pip/blob/8bf5731b84907bd1b41846334e98530bbe0cceaf/src/pip/_internal/utils/unpacking.py#L238
* zip: KO: https://github.com/pypa/pip/blob/8bf5731b84907bd1b41846334e98530bbe0cceaf/src/pip/_internal/utils/unpacking.py#L154
* https://github.com/python/cpython/blob/3.8/Lib/tarfile.py#L2281

setuptools:
* OK (shutil.copystat): https://github.com/pypa/setuptools/blob/990237c64404876791dc5985e7cfd260852dac0e/setuptools/archive_util.py#L88

So according to what I read:
* KO: pip install https://github.com/asciimoo/searx_external_plugin_example/archive/master.zip
* OK: pip install https://github.com/asciimoo/searx_external_plugin_example/archive/master.tar.gz <-- I don't know why it is KO.
* KO: in searx_external_plugin_example git clone (mtime)

Reply:
> The only consequence I see : it shows publicly when the plugin was installed (the reverse proxy will pick mtime).

Why is this a problem?

Reply:
It shows the installation time of the plugin, I don't think it is a problem.